flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlFoldFunction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.ttl;
+
+import org.apache.flink.api.common.functions.FoldFunction;
+
+/**
+ * This class wraps folding function with TTL logic.
+ *
+ * @param <T> Type of the values folded into the state
+ * @param <ACC> Type of the value in the state
+ *
+ * @deprecated use {@link TtlAggregateFunction} instead
+ */
+@Deprecated
+class TtlFoldFunction<T, ACC>
+	extends AbstractTtlDecorator<FoldFunction<T, ACC>>
+	implements FoldFunction<T, TtlValue<ACC>> {
+	TtlFoldFunction(FoldFunction<T, ACC> original, TtlConfig config, TtlTimeProvider timeProvider) {
+		super(original, config, timeProvider);
+	}
+
+	@Override
+	public TtlValue<ACC> fold(TtlValue<ACC> accumulator, T value) throws Exception {
+		return wrapWithTs(original.fold(getUnexpried(accumulator), value));
```

Comment:
I think that (similar to`TtlAggregationFunction`) you need to intercept `null` values of accumulator here and replace them by the default value.

Reply:
I this is true, that is a valuable case to be tested for all the appending states that they work correctly.

Reply:
As I understand, the wrapped states should already provide the default values. My idea was to wrap the original default value [in TTL factory](https://github.com/apache/flink/pull/6196/commits/99994dedb9a20244a2addd337617778b17fe8349#diff-13011fbe7c28b56b994783572b461aaeR174) with expiration timestamp `Long.MAX_VALUE`, basically never expiring. Good point about test cases for it, I will add them for appending states.

Reply:
It should be covered with `updateExpired` in `testExactExpirationOnWrite`